subsys/net/lib/sockets/getaddrinfo.c

```diff
@@ -20,12 +20,14 @@ LOG_MODULE_REGISTER(net_sock_addr, CONFIG_NET_SOCKETS_LOG_LEVEL);
 #include <net/socket_offload.h>
 #include <syscall_handler.h>
 
-#define AI_ARR_MAX	2
-
 #if defined(CONFIG_DNS_RESOLVER) || \
 	defined(CONFIG_NET_IPV6) || defined(CONFIG_NET_IPV4)
 #define ANY_RESOLVER
 
+#if !defined(CONFIG_DNS_RESOLVER)
+#define CONFIG_DNS_RESOLVER_AI_MAX_ENTRIES 1
```

Comment:
At some point in the past, there was some checks that prevented the code to define configuration options starting with `CONFIG_....`. Anyway, this line is not really needed if the check is added to `AI_ARR_MAX` define like this
```
#if defined(CONFIG_DNS_RESOLVER_AI_MAX_ENTRIES)
#define AI_ARR_MAX CONFIG_DNS_RESOLVER_AI_MAX_ENTRIES
#else
#define AI_ARR_MAX 1
#endif

Reply:
thanks for the suggestion, done.